### PR TITLE
Text search also searches specialities

### PR DESCRIPTION
**Pull request by nnahgodspower122**
updated developers model to include specialties

### CODE REVIEW

File: app/models/developer.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ class Developer < ApplicationRecord
   has_one :referring_user, through: :user
   has_many :conversations, -> { visible }
   has_many :messages, -> { where(sender_type: Developer.name) }, through: :conversations
+  has_many :specialty, through: :specialty_tags
```
Comment (bschrag620) [inline app/models/developer.rb:27]: I think this should be pluralized. You may also need to supply a `source`. I'm a bit rusty on the exact association definition here.
```
has_many :specialties, through: :specialty_tags, source: :specialty
```

Comment (joemasilotti) [reply app/models/developer.rb:27]: I'm not sure why this is needed at all. Up at the top of this file we have `HasSpecialties` which contains `has_many :specialties, -> { visible }, through: :specialty_tags, dependent: :destroy`.

Can you please remove this line and update the query to use the existing relationship?

File: app/models/developer.rb [hunk 2]
```diff
@@ -43,7 +44,7 @@ class Developer < ApplicationRecord
   validates :name, presence: true
   validates :response_rate, numericality: {greater_than_or_equal_to: 0, less_than_or_equal_to: 100}
 
-  pg_search_scope :filter_by_search_query, against: [:bio, :hero], using: {tsearch: {tsvector_column: :textsearchable_index_col}}
+  pg_search_scope :filter_by_search_query, against: [:bio, :hero], associated_against: {specialty: :name}, using: {tsearch: {tsvector_column: :textsearchable_index_col, prefix: true}}
```
Comment (joemasilotti) [inline app/models/developer.rb:47]: Nice! Do we need to update the index at all for this work?

**Review by joemasilotti — CHANGES_REQUESTED** [review]

File: app/models/developer.rb (hunk 2 above)
Comment (nnahgodspower122) [reply app/models/developer.rb:47]: hi @joemasilotti  I dont think we need to update the index

Comment (joemasilotti) [reply app/models/developer.rb:47]: Excellent! Thanks for confirming.

**Review by joemasilotti — APPROVED** [review]
Perfect!

**Comment by joemasilotti** [comment]
Thanks for the contribution, @nnahgodspower122!

Can you [email me](mailto:joe@railsdevs.com) and let me know how you'd like to receive the bounty? I can do GitHub Sponsorships, Venmo, PayPal, Zelle, and Wise.

**Pull request merged**